daemon/cluster/executor/container/container.go

```diff
@@ -77,8 +77,18 @@ func (c *containerConfig) name() string {
 		return c.task.Annotations.Name
 	}
 
-	// fallback to service.slot.id.
-	return strings.Join([]string{c.task.ServiceAnnotations.Name, fmt.Sprint(c.task.Slot), c.task.ID}, ".")
+	// fallback to service.slot.id
+	return fmt.Sprintf("%s.%s", c.shortName(), c.task.ID)
```

Comment:
@dnephin you are looking at `name()` which is infact `service.slot.taskid`. whereas the shortName is `service.slot` which is the task-name and it is used as an net-alias.

In short, yes a container will have container-id and task-name as net-aliases. Ofcourse service-name is used for load-balancing.